Lib/json/encoder.py

```diff
@@ -244,10 +244,13 @@ def floatstr(o, allow_nan=self.allow_nan,
             return text
 
 
-        if (_one_shot and c_make_encoder is not None
-                and self.indent is None):
+        if _one_shot and c_make_encoder is not None:
+            if self.indent is not None and not isinstance(self.indent, str):
```

Comment:
We can simplify this condition:

```suggestion
            if isinstance(self.indent, int):
```

Reply:
Your suggestion is good, but there might be a few corner cases. What about values for `self.indent`that are not an instance of `int`, but do support `indent = ' ' * self.indent`?

I am ok with making the change, but would like the opinion of a core developer on this.

Reply:
In the Python implementation below:
```py
    if _indent is not None and not isinstance(_indent, str):
        _indent = ' ' * _indent
```
We should use the same code in both cases. Actually, we can move this code outside of `_make_iterencode()`, since it is used in both implementations. Unless you want to move it in C for the C implementation.

Reply:
You can also negate the condition:

```python
if self.indent is None or isinstance(self.indent, str):
    indent = self.indent
else:
    indent = ' ' * self.indent
```

Reply:
You can resolve this.